From 9d89a073b46ad7ebdfd88963d36b29e920aebcd9 Mon Sep 17 00:00:00 2001
From: Xavier Hanin <xavier@apache.org>
Date: Tue, 1 May 2007 11:30:06 +0000
Subject: [PATCH] FIX: ivy:install ant task does not fail on error (IVY-475)
 use haltonfailure instead of haltonunresolved, and halt in case of any
 failure

git-svn-id: https://svn.apache.org/repos/asf/incubator/ivy/core/trunk@534012 13f79535-47bb-0310-9956-ffa450edef68
---
 src/java/org/apache/ivy/ant/IvyInstall.java      | 15 +++++++--------
 test/java/org/apache/ivy/ant/IvyInstallTest.java |  5 +----
 2 files changed, 8 insertions(+), 12 deletions(-)

diff --git a/src/java/org/apache/ivy/ant/IvyInstall.java b/src/java/org/apache/ivy/ant/IvyInstall.java
index 020d71a33..97ed87d6f 100644
--- a/src/java/org/apache/ivy/ant/IvyInstall.java
+++ b/src/java/org/apache/ivy/ant/IvyInstall.java
@@ -46,7 +46,7 @@ public class IvyInstall extends IvyTask {
     private boolean _transitive;
     private String _type;
     private String _matcher = PatternMatcher.EXACT;
-    private boolean _haltOnUnresolved = true;
+    private boolean _haltOnFailure = true;
     
     public void execute() throws BuildException {
         Ivy ivy = getIvyInstance();
@@ -81,17 +81,16 @@ public void execute() throws BuildException {
             throw new BuildException("impossible to install "+ mrid +": "+e, e);
         }
         
-        if (report.getUnresolvedDependencies().length > 0 && isHaltonunresolved()) {
-            throw new BuildException(report.getUnresolvedDependencies().length
-            		+" unresolved dependencies - see output for details");
+        if (report.hasError() && isHaltonfailure()) {
+            throw new BuildException("Problem happened while installing modules - see output for details");
         }
     }
     
-    public boolean isHaltonunresolved() {
-        return _haltOnUnresolved;
+    public boolean isHaltonfailure() {
+        return _haltOnFailure;
     }
-    public void setHaltonunresolved(boolean haltOnUnresolved) {
-        _haltOnUnresolved = haltOnUnresolved;
+    public void setHaltonfailure(boolean haltOnFailure) {
+        _haltOnFailure = haltOnFailure;
     }
     
     public File getCache() {
diff --git a/test/java/org/apache/ivy/ant/IvyInstallTest.java b/test/java/org/apache/ivy/ant/IvyInstallTest.java
index a2bdf6476..ede0181d4 100644
--- a/test/java/org/apache/ivy/ant/IvyInstallTest.java
+++ b/test/java/org/apache/ivy/ant/IvyInstallTest.java
@@ -78,9 +78,6 @@ public void testDependencyNotFoundFailure() {
             fail("unknown dependency, failure expected (haltunresolved=true)");
         } catch (BuildException be) {
             // success
-            assertTrue("invalid exception message, it should contain '1 unresolved',"
-            		+" but it's: '"+be.getMessage()+"'", 
-            		be.getMessage().indexOf("1 unresolved") != -1);
         }  
     }
 
@@ -90,7 +87,7 @@ public void testDependencyNotFoundSuccess() {
         _install.setRevision("zzz");
         _install.setFrom("test");
         _install.setTo("1");
-        _install.setHaltonunresolved(false);
+        _install.setHaltonfailure(false);
         
         try {
             _install.execute();
